README.md

```diff
@@ -13,3 +13,39 @@
   complexity suggests a different time.
 
 Add your answers to this markdown file.
+
+Bullet 1
+
+1. Asymptotic analysis is theoretical, as it describes algorithm behavior with a set of unspecific inputs.  Actual performance is empirical, as it describes algorithm behavior with a specific set of inputs.  The two are different, as actual performance depends on constants and machine performance, whereas asymptotic analysis ignores those factors.
+
+
+2. Actual performance is affected by hardware, software, and the characteristics of the input data, whereas asymptotic analysis is primarily affected by the input size.  Asymptotic analysis is also affected by the characteristics of the input data, but in a theoretical sense, as it often analyzes best, worst, and average cases based on predefined assumptions about input behavior that don’t always align with real-world data patterns.  In comparison, actual performance takes into account real-world input characteristics such as specific distributions, clustering, or ordering of data.  Combining these real-world factors with hardware and software, they affect actual performance much more than the effect of the theoretical sense of input data characteristics on asymptotic analysis, which is why asymptotic analysis is primarily affected by the input size.
+
+3. In addition to ignoring constants, asymptotic analysis also ignores lower-order terms, which can be misleading when compared to actual performance since when lower-order terms are accounted for, they can have a decent effect on the runtime.
+
+Bullet 2
+
+The time complexity depends on if the binary search tree is balanced or not.  If it is balanced, it has a best case of O(1) and an average and worst case of O(log(n)).  If it isn’t balanced, it has a best case of O(1), and an average and worst case of O(n).
```

Comment:
Can you explain this a bit more please? There's no tree -- binary search simply cuts the list in half. How can this be unbalanced?